Documentation/ceph-debugging.md

```diff
@@ -0,0 +1,231 @@
+    - [Set Ceph log levels](#set-ceph-log-levels)
+    - [Get container logs](#get-container-logs)
+    - [Describe Deployments, Pods, and other Kubernetes resources](#describe-deployments-pods-and-other-kubernetes-resources)
+    - [Stop the operator](#stop-the-operator)
+    - [Restart the operator](#restart-the-operator)
+    - [Restart a daemon pod](#restart-a-daemon-pod)
+    - [Determine the node a Pod runs on](#determine-the-node-a-pod-runs-on)
+  - [Debugging](#debugging)
+    - [General](#general)
+    - [OSDs](#osds)
+    - [Mons](#mons)
+    - [Problems with user applications mounting Rook Storage Classes (CSI)](#problems-with-user-applications-mounting-rook-storage-classes-csi)
+
+## Methods
+There are a number of basic actions a user might need to take during debugging. These actions are
+defined here for reference when they are mentioned in more documentation below.
+
+> **NOTE:** Please note that this document is not devoted to an in-depth explanation of what
+> Kubernetes is, what its features are, how it is used, how to navigate it, or how to debug
+> applications that run on it. This document will use Kubernetes terms, and users are expected to be
+> empowered to look up Kubernetes information they do not already have. This document will connect
+> dots to explain how to use Kubernetes tools to get information needed in the Rook-Ceph context and
+> (when relevant) will briefly explain how Rook uses Kubernetes features.
+
+### Set operator log level to debug
+In general, the first place to look when encountering a failure is to get logs for the rook-ceph-operator pod. To get the best logs possible, set the operator log level to "DEBUG". There are many ways to do this, including modifying Helm's values.yaml or modifying the operator.yaml manifest. Regardless of method chosen, the log level can always be set by editing the deployment directly with kubectl.
+
+```sh
+kubectl --namespace rook-ceph edit deployment rook-ceph-operator
+# manually edit ROOK_LOG_LEVEL 'env' value "DEBUG" (more logs) or "INFO" (less logs) as desired
+```
+
+After editing the deployment, the operator pod will restart automatically and will start outputting
+logs with the new log level.
+
+If you are experiencing a particular failure, it may take some time for the Rook operator to reach
+the failure location again to report debug logs.
+
+### Use the toolbox pod
+Use the Rook toolbox pod to interface directly with the Ceph cluster via CLI. Enter into the toolbox
+pod as shown below.
+```
+kubectl --namespace rook-ceph exec -it deploy/rook-ceph-tools -- bash
+```
+
+If the rook-ceph-tools deployment does not exist, it should be created using the toolbox.yaml manifest.
+
+### Set Ceph log levels
+To set log levels for Ceph daemons, it is advised to use the Ceph CLI from the toolbox pod.
+
+### Get container logs
+Getting logs for a container in Kubernetes can be fairly easy, but to get logs with more clarity, it
```

Comment:
Majority of the time we just need the logs for the main container, right? How about separating this by "main logs" and "logs for all containers"? Instead of saying one is quick and dirty and the other is slow and precise. :)